Resolve remaining TODOs (#4)
kavon/swift-evolution: proposals/NNNN-noncopyable-generics.md

let u: any Pizza & ~Copyable = ... // signature <Self: Pizza>
let _: any ~Copyable = u.peelOneTopping() // signature <Self>
```
Associated types within a protocol erased to an existential preserve conformances to `Copyable`.
So when calling `peelOneTopping` on an `any Pizza`, an `any ~Copyable` value is returned instead of `any Copyable`.

### Diagnosing contradictory intent

> An error will be diagnosed whenever a type expression includes an explicit `~Copyable` but the type resolves to be `Copyable`.

Reviewer: ```suggestion
An error will be diagnosed whenever a type includes an explicit `~Copyable` but its generic requirements can only be satisfied if it were `Copyable`.
```
